lib/govuk_schemas/random_helpers.rb

```diff
@@ -0,0 +1,126 @@
+module GovukSchemas
+  # @private
+  class RandomHelpers
+    WORDS = %w[Lorem ipsum dolor sit amet consectetur adipiscing elit. Ut suscipit at mauris non bibendum. Ut ac massa est. Aenean tempor imperdiet leo vel interdum. Nam sagittis cursus sem ultricies scelerisque. Quisque porttitor risus vel risus finibus eu sollicitudin nisl aliquet. Sed sed lectus ac dolor molestie interdum. Nam molestie pellentesque purus ac vestibulum. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Suspendisse non tempor eros. Mauris eu orci hendrerit volutpat lorem in tristique libero. Duis a nibh nibh.].freeze
+
+    def initialize(random: Random.new)
+      @random = random
+    end
+
+    def string_for_type(type)
+      if type == "date-time"
+        time
+      elsif type == "uri"
+        uri
+      else
+        raise "Unknown attribute type `#{type}`"
+      end
+    end
+
+    def time
+      seconds_ago = @random.rand(-5_000_000..4_999_999)
+      (Time.now + seconds_ago).iso8601
```

Comment:
I guess this Time.now causes an element of unpredictability, which explains why you need the timecop gem installed. I think it'd probably best to cut it out as you lose the deterministic approach.

Simplest approach I came up with as a replacement would be `Time.new(2012, 2, 1) + @random.rand(0..500_000_000)` where you'll get a time from GOV.UK's birthday until 2027.

There's the over thinker in me thinking that you probably don't actually want this as times in the future (despite the current code) and you'd want this to not date oddly, so you could use the current year in it so this only changes once a year:

```
irb(main):011:0> year = Date.today.year.then { |y| rand((y - 5)..y) }
=> 2017
irb(main):012:0> Time.new(year) - rand(0..10_000_000)
=> 2016-10-01 23:58:05 +0100
```

Which would give you a time within roughly the last 4 previous years deterministically (unless you ran it at stroke of midnight on new years eve).

Reply:
Ah, I like this, because otherwise gh-pages would still have a handful of line changes for every schema on every commit 😱 I was wondering how we'd deal with that, but was going to tackle it later. May as well tackle it now!
I'll have a go at implementing your suggestion.